docs/sphinx/examples/matrix_multiplication.rst

```diff
@@ -0,0 +1,107 @@
+..
+    Copyright (C) 2021 Dimitra Karatza
+
+    SPDX-License-Identifier: BSL-1.0
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _examples_matrix_multiplication:
+
+===================
+Parallel algorithms
+===================
+
+This program will perform a matrix multiplication in parallel. The output will look something like this:
+
+.. code-block:: text
+
+   Matrix A is : 
+   4 9 6 
+   1 9 8 
+
+   Matrix B is : 
+   4 9 
+   6 1 
+   9 8 
+
+   Resultant Matrix is : 
+   124 93 
+   111 127 
+
+Setup
+=====
+
+The source code for this example can be found here:
+:download:`matrix_multiplication.cpp
+<../../examples/quickstart/matrix_multiplication.cpp>`.
+
+To compile this program, go to your |hpx| build directory (see
+:ref:`hpx_build_system` for information on configuring and building |hpx|) and
+enter:
+
+.. code-block:: shell-session
+
+   $ make examples.quickstart.matrix_multiplication
+
+To run the program type:
+
+.. code-block:: shell-session
+
+   $ ./bin/matrix_multiplication
```

Comment:
I think it'd be nice if this took the size of the matrices as command line options. The defaults can very well be 2x3 and 3x2 so you can leave the line above as it is, but if someone wants to play with the example it'd be nice to be able to change the sizes.

Defining three command-line options, `--n`, `--m`, `--k`, for the dimensions should be ok, where the first matrix is `n` x `m` and the second `m` x `k`.